addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRuleUnitTest.java

```diff
@@ -0,0 +1,217 @@
+        assertThat(alertsRaised, hasSize(1));
+        Alert alert = alertsRaised.get(0);
+        assertEquals("Bypassing 403", alert.getName());
+        assertEquals(Alert.RISK_MEDIUM, alert.getRisk());
+        assertEquals(Alert.CONFIDENCE_MEDIUM, alert.getConfidence());
+    }
+
+    @Test
+    public void shouldAlertIfOkWithHeaderPayload2() throws Exception {
+        // Given
+        nano.addHandler(new ForbiddenResponse(PROTECTED_PATH));
+        nano.addHandler(new HeaderResponse1("/anything"));
+        HttpMessage msg = this.getHttpMessage(PROTECTED_PATH);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        Alert alert = alertsRaised.get(0);
+        assertEquals("Bypassing 403", alert.getName());
+        assertEquals(Alert.RISK_MEDIUM, alert.getRisk());
+        assertEquals(Alert.CONFIDENCE_MEDIUM, alert.getConfidence());
+    }
+
+    @Test
+    public void shouldAlertIfOkWithHeaderPayload3() throws Exception {
+        // Given
+        nano.addHandler(new ForbiddenResponse(PROTECTED_PATH));
+        nano.addHandler(new HeaderResponse1("/"));
+        HttpMessage msg = this.getHttpMessage(PROTECTED_PATH);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        Alert alert = alertsRaised.get(0);
+        assertEquals("Bypassing 403", alert.getName());
+        assertEquals(Alert.RISK_MEDIUM, alert.getRisk());
+        assertEquals(Alert.CONFIDENCE_MEDIUM, alert.getConfidence());
+    }
+
+    @Test
+    public void shouldAlertIfOkWithHeaderPayload4() throws Exception {
+        // Given
+        nano.addHandler(new HeaderResponse2(PROTECTED_PATH));
+        HttpMessage msg = this.getHttpMessage(PROTECTED_PATH);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        Alert alert = alertsRaised.get(0);
+        assertEquals("Bypassing 403", alert.getName());
+        assertEquals(Alert.RISK_MEDIUM, alert.getRisk());
+        assertEquals(Alert.CONFIDENCE_MEDIUM, alert.getConfidence());
+    }
+
+    private static class OkResponse extends NanoServerHandler {
+
+        public OkResponse(String path) {
+            super(path);
+        }
+
+        @Override
+        protected Response serve(IHTTPSession session) {
+            return newFixedLengthResponse(Response.Status.OK, "text/html", GENERIC_RESPONSE);
+        }
+    }
+
+    private static class ForbiddenResponse extends NanoServerHandler {
+
+        public ForbiddenResponse(String path) {
+            super(path);
+        }
+
+        @Override
+        protected Response serve(IHTTPSession session) {
+            return newFixedLengthResponse(Response.Status.FORBIDDEN, "text/html", GENERIC_RESPONSE);
+        }
+    }
+
+    private static class HeaderResponse1 extends NanoServerHandler {
```

Comment:
Same here regarding using specific names, although this should be changed to allow to pass the header being tested.

Reply:
Note the last part, it should be changed to allow to pass the header (to avoid duplicating the same code for each header).